src/jit/gentree.cpp

```diff
@@ -12120,14 +12120,57 @@ GenTree* Compiler::gtFoldExprSpecial(GenTree* tree)
 
     val = cons->gtIntConCommon.IconValue();
 
+    // Transforms that would drop op cannot be performed if op has side effects
+    bool opHasSideEffects = (op->gtFlags & GTF_SIDE_EFFECT) != 0;
+
+    // Helper function that creates a new IntCon node and morphs it, if required
+    auto NewMorphedIntConNode = [&](int value) -> GenTreeIntCon* {
+        GenTreeIntCon* icon = gtNewIconNode(value);
+        if (fgGlobalMorph)
+        {
+            fgMorphTreeDone(icon);
+        }
+        return icon;
+    };
+
     /* Here op is the non-constant operand, val is the constant,
        first is true if the constant is op1 */
```

Comment:
This part of this comment doesn't make sense:
`first is true if the constant is op1 `

Reply:
Heh, I guess there used to be a variable named `first` here. Not my comment but I'll fix this together with the rest of comments.